src/loader/loader.c

```diff
@@ -349,7 +349,7 @@ post_analyze_hook(ParseState *pstate, Query *query)
 				}
 				break;
 			case T_DropOwnedStmt:
-				if (drop_owned_statement_drops_extension((DropOwnedStmt *) query->utilityStmt))
+				if (extension_exists() && drop_owned_statement_drops_extension((DropOwnedStmt *) query->utilityStmt))
```

Comment:
I think this check should probably be at the top of this hook function since the hook should not do anything at all if the extension is not loaded. I.e., there's no reason to stop DB workers or dropping the extension either in that case. If, for some reason, that would not work, the check should probably be part of the `drop_owned_statment_drops_extension()` function.

Reply:
That seems indeed to be the case. Then I'd move the `extension_exists()` check into both the extension drop and drop owned functions.

Reply:
adjusted and added comment